javers-core/src/main/java/org/javers/snapshots-compiler/JaversSnapshotsCompiler.java

```diff
@@ -0,0 +1,216 @@
+        //Optional<CdoSnapshot> latestSnapshot = javers.getLatestSnapshot()getLatestSnapshot(convertInstanceIdToDTO(instanceId));
+        return latestSnapshot.isPresent() ? compileEntityStateFromSnapshot(latestSnapshot.get()) : null;
+    }
+
+    /**
+     * Load latest entity state for entity
+     *
+     * @param entity
+     * @return
+     */
+    public Object compileLatestEntityStateForEntity(Object entity) {
+        return compileLatestEntityState(javers.idBuilder().instanceId(entity));
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private Object compileEntityInternal(CdoSnapshot snap) throws IllegalAccessException, InstantiationException {
+
+        if (idToEntityMap.containsKey(((InstanceId) snap.getGlobalId()).getCdoId()))
+            return idToEntityMap.get(((InstanceId)snap.getGlobalId()).getCdoId());
+
+        Class clientsClass = snap.getManagedType().getBaseJavaClass();
+
+        Object instance = clientsClass.newInstance();
+        idToEntityMap.put(((InstanceId)snap.getGlobalId()).getCdoId(), instance);
+        //for (Property property : snap.getState().getProperties()) {
+        for (String propertyName : snap.getState().getProperties()) {
+            Object propertyValue = snap.getPropertyValue(propertyName);
+            Object propertyValueToSet = propertyValue;
+            if (propertyValue != null) {
+                try {
+                    Method setter = getSetterForProperty(clientsClass, propertyName);
+                    Class<?> targetType = setter.getParameterTypes()[0];
+                    if (targetType.isAssignableFrom(java.util.List.class)
+                            || targetType.isAssignableFrom(java.util.Set.class)) {
+                        Collection collection = (Collection) propertyValue;
+                        Set toBeDeleted = new HashSet<>();
+                        for (Object element : collection.toArray()){
+                            if (element instanceof InstanceId) {
+                                InstanceId instanceId = (InstanceId) element;
+
+                                Object targetObject = compileEntityForCommitId(
+                                        instanceId, snap.getCommitId());
+                                collection.add(targetObject);
+                                toBeDeleted.add(element);
+                            }
+                        }
+                        for (Object element : toBeDeleted) {
+                            collection.remove(element);
+                        }
+                    } else if (propertyValue instanceof InstanceId) {
+                        propertyValueToSet = compileEntityForCommitId(
+                                (InstanceId) propertyValue, snap.getCommitId());
+                    }
+                    setter.invoke(instance, propertyValueToSet);
+                } catch (Exception exc) {
```

Comment:
swallowing all Exceptions is generally an antipatten, wrap it in JaversRuntimeException and rethrow